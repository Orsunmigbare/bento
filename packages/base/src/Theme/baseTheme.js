import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
