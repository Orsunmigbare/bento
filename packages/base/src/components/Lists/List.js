import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
`

const List = ({ children, className }) => (
  <ListUl className={className}>{children}</ListUl>
)

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element
  ]).isRequired,
  className: PropTypes.string
}

export default List
