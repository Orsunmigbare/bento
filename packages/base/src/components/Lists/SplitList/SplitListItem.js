import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  targetSize,
  spacing,
  borderRadius,
  fontFamily,
  font,
  color,
  themify
} from "@staccx/theme"

const ListItem = styled.li`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  border-bottom: 1px solid ${color.line};
  padding: ${spacing.tiny()}
    ${spacing.medium()};
  font-size: ${font.base};
  font-family: ${fontFamily.body()};

  &:last-child {
    border-bottom: none;
  }
`

const SplitListItem = ({ children, className }) => {
  if (React.Children.count(children) > 2) {
    console.warn(
      "SplitListItem exceeded max children: Can only contain 2 children. Others will be omitted."
    )
  }
  return (
    <ListItem className={className}>
      {React.Children.map(children, (child, index) => {
        return index <= 1 ? <span>{child}</span> : null
      })}
    </ListItem>
  )
}

SplitListItem.defaultProps = {
  className: ""
}

SplitListItem.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.array,
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  className: PropTypes.string
}

export default SplitListItem
