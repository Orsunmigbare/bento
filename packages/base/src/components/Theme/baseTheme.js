export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
