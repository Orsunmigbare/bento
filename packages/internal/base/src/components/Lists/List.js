import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { themify, themeProps } from "@staccx/theme"

const List = ({ children, className, ...restProps }) => (
  <ListUl className={className} {...restProps}>
    {children}
  </ListUl>
)

export const LIST = "LIST"

export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${themify(LIST)};
`

List.defaultProps = {
  className: ""
}

List.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string
}

export default List
