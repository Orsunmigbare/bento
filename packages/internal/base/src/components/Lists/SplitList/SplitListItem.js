import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  spacing,
  fontFamily,
  font,
  color,
  themify,
  themeProps
} from "@staccx/theme"

const SplitListItem = ({ children, className, ...restProps }) => {
  if (React.Children.count(children) > 2) {
    console.warn(
      "SplitListItem exceeded max children: Can only contain 2 children. Others will be omitted."
    )
  }
  return (
    <ListItem className={className} {...restProps}>
      {React.Children.map(children, (child, index) => {
        return index <= 1 ? <span>{child}</span> : null
      })}
    </ListItem>
  )
}

export const SPLIT_LIST = "split_list"
const ListItem = styled.li`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  border-bottom: 1px solid ${color.line};
  padding: ${spacing.tiny()} ${spacing.medium()};
  font-size: ${font.base};
  font-family: ${fontFamily.body()};

  &:last-child {
    border-bottom: none;
  }
  ${themify(SPLIT_LIST)};
`

SplitListItem.defaultProps = {
  className: ""
}

SplitListItem.propTypes = {
  children: themeProps.children.isRequired,
  className: PropTypes.string
}

export default SplitListItem
