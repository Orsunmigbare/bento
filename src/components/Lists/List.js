import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, commonPropTypes, themePropTypes } from "../../theming"

const List = ({ children, className, ordered, ...restProps }) => {
  if (!ordered) {
    return (
      <ListUl className={className} {...restProps}>
        {children}
      </ListUl>
    )
  }
  return (
    <ListOl className={className} {...restProps}>
      {children}
    </ListOl>
  )
}

List.themeProps = {
  list: {
    name: "LIST",
    description: "",
    type: themePropTypes.style
  }
}
export const ListUl = styled.ul`
  list-style-type: none;
  margin-left: 0;
  padding-left: 0;
  ${applyVariants(List.themeProps.list)};
`

export const ListOl = ListUl.withComponent("ol")

List.defaultProps = {
  className: "",
  ordered: false
}

List.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string,
  /**
   * Prop to set ordered or unordered List (ul/ol)
   */
  ordered: PropTypes.bool
}

export default List
