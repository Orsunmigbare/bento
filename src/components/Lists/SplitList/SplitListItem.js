import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import {
  spacing,
  fontFamily,
  font,
  color,
  applyVariants,
  commonPropTypes,
  themePropTypes
} from "../../../theming"

const SplitListItem = ({ children, className, ...restProps }) => {
  if (React.Children.count(children) > 2) {
    console.warn(
      "SplitListItem exceeded max children: Can only contain 2 children. Others will be omitted."
    )
  }
  return (
    <ListItem className={className} {...restProps}>
      {React.Children.map(children, (child, index) => {
        return index <= 1 ? <span>{child}</span> : null
      })}
    </ListItem>
  )
}

SplitListItem.themeProps = {
  listItem: {
    name: "split_list",
    description: "",
    type: themePropTypes.style
  }
}

const ListItem = styled.li`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  border-bottom: 1px solid ${color.line};
  padding: ${spacing.tiny} ${spacing.medium};
  font-size: ${font.base};
  font-family: ${fontFamily.body()};

  &:last-child {
    border-bottom: none;
  }
  ${applyVariants(SplitListItem.themeProps.listItem)};
`

SplitListItem.defaultProps = {
  className: ""
}

SplitListItem.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string
}

export default SplitListItem
