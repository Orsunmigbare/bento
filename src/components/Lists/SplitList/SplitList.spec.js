import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import SplitListItem from "./SplitListItem"
import List from "../List"

const myList = [
  {
    myUniqueId: "221e98j",
    myLeftProp: "Amount",
    myRightProp: 500
  },
  {
    myUniqueId: "addw23",
    myLeftProp: "Amount",
    myRightProp: 1500
  },
  {
    myUniqueId: "h213",
    myLeftProp: "Has three children!",
    myRightProp: "500",
    myThirdProp: "1500"
  }
]

describe("splitListItem", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              {myList.map(listItem => (
                <SplitListItem key={listItem.myUniqueId}>
                  {listItem.myLeftProp}
                  {listItem.myRightProp}
                </SplitListItem>
              ))}
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("More than 2 children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <List>
              {myList.map(listItem => (
                <SplitListItem key={listItem.myUniqueId}>
                  {listItem.myLeftProp}
                  {listItem.myRightProp}
                  {listItem.myLeftProp}
                  {listItem.myRightProp}
                </SplitListItem>
              ))}
            </List>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
