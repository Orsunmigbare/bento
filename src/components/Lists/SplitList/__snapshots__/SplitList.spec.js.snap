// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitListItem Snapshots More than 2 children 1`] = `
<ul
  className="sc-AykKD jpnaqQ"
>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Amount
    </span>
    <span>
      500
    </span>
  </li>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Amount
    </span>
    <span>
      1500
    </span>
  </li>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Has three children!
    </span>
    <span>
      500
    </span>
  </li>
</ul>
`;

exports[`splitListItem Snapshots No props 1`] = `
<ul
  className="sc-AykKD jpnaqQ"
>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Amount
    </span>
    <span>
      500
    </span>
  </li>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Amount
    </span>
    <span>
      1500
    </span>
  </li>
  <li
    className="sc-AykKC doyRde"
  >
    <span>
      Has three children!
    </span>
    <span>
      500
    </span>
  </li>
</ul>
`;
